starfish/experiment/experiment.py

```diff
@@ -89,10 +89,11 @@ def name(self) -> str:
     def image_types(self) -> Set[str]:
         return set(self._images.keys())
 
-    def __getitem__(self, item) -> ImageStack:
-        if isinstance(self._images[item], TileSet):
-            self._images[item] = ImageStack.from_tileset(self._images[item])
-        return self._images[item]
+    def get_image(self, item: str, crop_parameters: Optional[CropParameters]=None) -> ImageStack:
+        return ImageStack.from_tileset(self._images[item], crop_parameters)
+
+    def __getitem__(self, item: str) -> ImageStack:
```

Comment:
`TileSet` should not have a `to_imagestack` method.  For most users, I don't think they would ever care about instantiating a TileSet.  In any case, I think the point is somewhat moot if we move to dask, because instantiating a dask array has a much lower cost.